File: 409.LongestPalindrome/Longest.java
class Solution {
    public int longestPalindrome(String s) {
        Map<Character, Integer> count = new HashMap<>();
        for (char c : s.toCharArray()) {
            if (!count.containsKey(c)) {
                count.put(c, 0);
            }
            count.put(c, count.get(c)+1);
        }

        int sum = 0;
        int odd = 0;
        for (int v : count.values()) {
            sum += v;
            if (v % 2 != 0) {
                odd++;
            }
        }

        return sum - odd + Math.min(odd, 1);
    }
}

File: 409.LongestPalindrome/longest.cc
class Solution {
public:
    int longestPalindrome(string s) {
        map<char,int> count;
        for (char c : s) {
            count[c]++;
        }
        int sum = 0, odd = 0;
        for (map<char,int>::iterator it = count.begin(); it != count.end(); ++it) {
            int v = it->second;
            sum += v;
            if (v % 2) {
                odd++;
            }
        }

        return sum - odd + min(odd, 1);
    }
};
